presto-parser/src/main/java/com/facebook/presto/sql/tree/SetOperation.java

```diff
@@ -13,19 +13,44 @@
  */
 package com.facebook.presto.sql.tree;
 
+import java.util.List;
 import java.util.Optional;
 
 public abstract class SetOperation
         extends QueryBody
 {
-    protected SetOperation(Optional<NodeLocation> location)
+    private final Type type;
+    private final boolean distinct;
+
+    public enum Type
```

Comment:
No, I'm suggesting getting rid of it altogether. It duplicates information encoded in the class hierarchy.